### PR TITLE
Add support to Adaptiope dataset for cross domain ssl

### PR DESCRIPTION
**Pull request by Momofil31**
## What does this PR do?
This pull request adds support to Adaptiope [^fn2] dataset, in order to train and evaluate semi-supervised learning algorithms in a scenario where labeled and unlabeled data belong to different data distributions. 
This SSL scenario is known to be challenging since "When the distributions of labeled and unlabeled data do not belong to the same one [...], the more unlabeled data is utilized, the worse the performance will be." (Yang et al, 2021)[^fn1].

Adaptiope is a domain adaptation dataset for image classification composed of three domains (product images, real life images, and synthetic renders). For each domain, there are 123 classes with 100 samples per class. 

I do not have enough computational resources to perform extensive experiments, however, I managed to perform some preliminary ones with the configs added in this pull request. These result are purely indicative since I didn't perform any hyperparameter search and trained for very few iterations.
More specifically, I finetuned a pre-trained ViT Small for 20480 iterations with 615 labels (5% supervision) on product domain as supervised set and 12300 unlabeled samples from real life domain.
| Method | Accuracy |
| --- | --- |
| _Supervised lower bound (615 labels)_ | 78.17 % |
| _Fully Supervised upper bound (9840 labels)_ | 93.01% |
| FixMatch | 78.78 % |
| ReMixMatch | 87.24 % |

From these results we can notice that ReMixMatch does not suffer the domain shift, probably thanks to the distribution alignment module on the guessed labels, while FixMatch is not able to exploit the unlabeled data to increase its accuracy over the baseline.

More research is needed in SSL scenario, so I think it could be useful to integrate this feature in the USB  benchmarking library.

[^fn1]: Yang, X., Song, Z., King, I., & Xu, Z.. (2021). A Survey on Deep Semi-supervised Learning.
[^fn2]: Ringwald, T., & Stiefelhagen, R. (2021). Adaptiope: A Modern Benchmark for Unsupervised Domain Adaptation. In Proceedings of the IEEE/CVF Winter Conference on Applications of Computer Vision (WACV) (pp. 101-110).

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you list all the **breaking changes** introduced by this pull request?

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Comment by Momofil31** [comment]
@microsoft-github-policy-service agree

**Comment by qianlanwyd** [comment]
Hello, thanks a lot for your PR. We will have an update recently (For imbalanced SSL, Classical settings with training logs, etc.). After that, we will check your code and rerun it for providing training logs. If you have any suggestion/question, feel free to contact.

**Comment by Hhhhhhao** [comment]
@Momofil31 Thanks for this PR! Since the change is related to domain adaption and we will be working on openset ssl algorithms and other robust semi-supervised learning perspectives for the next update, this PR will be updated and merged after we decide the whole pipeline.
